packages/rs-drive-abci/src/abci/handler/load_snapshot_chunk.rs

```diff
@@ -0,0 +1,197 @@
+use crate::abci::app::{PlatformApplication, SnapshotManagerApplication};
+use bincode::{Decode, Encode};
+use drive::grovedb::GroveDb;
+use tenderdash_abci::proto::abci as proto;
+//use platform_version::version::PlatformVersion;
+use crate::abci::AbciError;
+use crate::error::Error;
+use crate::rpc::core::CoreRPCLike;
+use dpp::version::PlatformVersion;
+
+pub fn load_snapshot_chunk<A, C>(
+    app: &A,
+    request: proto::RequestLoadSnapshotChunk,
+) -> Result<proto::ResponseLoadSnapshotChunk, Error>
+where
+    A: SnapshotManagerApplication + PlatformApplication<C>,
+    C: CoreRPCLike,
+{
+    tracing::trace!(
+        "[state_sync] api load_snapshot_chunk height:{} chunk_id:{}",
+        request.height,
+        hex::encode(&request.chunk_id)
+    );
+    let matched_snapshot = app
+        .snapshot_manager()
+        .get_snapshot_at_height(&app.platform().drive.grove, request.height as i64)
+        .map_err(|_| {
+            AbciError::StateSyncInternalError(
+                "load_snapshot_chunk failed: error matched snapshot".to_string(),
+            )
+        })?
+        .ok_or_else(|| {
+            AbciError::StateSyncInternalError(
+                "load_snapshot_chunk failed: empty matched snapshot".to_string(),
+            )
+        })?;
+    let db = GroveDb::open(&matched_snapshot.path).map_err(|e| {
+        AbciError::StateSyncInternalError(format!(
+            "load_snapshot_chunk failed: error opening grove:{}",
+            e
+        ))
+    })?;
+    let chunk = db
+        .fetch_chunk(
+            &request.chunk_id,
+            None,
+            request.version as u16,
+            &PlatformVersion::latest().drive.grove_version,
+        )
+        .map_err(|e| {
+            AbciError::StateSyncInternalError(format!(
+                "load_snapshot_chunk failed: error fetching chunk{}",
+                e
+            ))
+        })?;
+
+    // wrap chunk data with some metadata
+    let chunk_data = ChunkData::new(&chunk).serialize()?;
+    let response = proto::ResponseLoadSnapshotChunk { chunk: chunk_data };
+    Ok(response)
+}
+
+fn crc32(data: &[u8]) -> [u8; 4] {
+    let crc32 = crc::Crc::<u32>::new(&crc::CRC_32_CKSUM);
+    let mut digest = crc32.digest();
+    digest.update(data);
+    digest.finalize().to_le_bytes()
+}
+
+// ChunkData wraps binary chunk data with additional metadata, like checksum and size.
+//
+// There is no way to update the chunk - create new ChunkData instead.
+//
+// TODO: Use Platform encoding instead of raw binencode?
+#[derive(Debug, Clone, Encode, Decode)]
+pub(crate) struct ChunkData {
```

Comment:
Not sure we still need it, feel free to remove if you think we don't.

Otherwise, I think this should be moved somewhere (just not sure where) and maybe refactored to match versioning approach.